Code/GraphMol/Chirality.cpp

```diff
@@ -1637,14 +1637,18 @@ void findPotentialStereoBonds(ROMol &mol, bool cleanIt) {
           !(mol.getRingInfo()->numBondRings((*bondIt)->getIdx()))) {
         // we are ignoring ring bonds here - read the FIX above
         Bond *dblBond = *bondIt;
-        // if the bond is flagged as EITHERDOUBLE, we ignore it:
+        // We ignore bonds flagged as EITHERDOUBLE or STEREOANY which have
+        // stereo atoms set.
         if (dblBond->getBondDir() == Bond::EITHERDOUBLE ||
-            dblBond->getStereo() == Bond::STEREOANY) {
+            (dblBond->getStereo() == Bond::STEREOANY &&
+             dblBond->getStereoAtoms().size() == 2)) {
           continue;
         }
-        // proceed only if we either want to clean the stereocode on this bond
-        // or if none is set on it yet
-        if (cleanIt || dblBond->getStereo() == Bond::STEREONONE) {
+        // proceed only if we either want to clean the stereocode on this bond,
+        // if none is set on it yet, or it is STEREOANY and we need to find
```

Comment:
What if it's EITHER and we need the adjacent atoms? would we run into the same bug?

Reply:
My two cents: given that EITHER is a direction and not a stereo value, I think it can be ignored. When we revisit the stereo representation I think it would behoove us to remove that direction option anyway.

Reply:
oh, of course.

Reply:
Yeah, that was my impression about the EITHER bond: if it is not marked as stereo, ignore it.